src/main/scala/com/codacy/rules/ReportRules.scala

```diff
@@ -194,18 +194,33 @@ class ReportRules(coverageServices: => CoverageServices) extends LogSupport {
     */
   private[rules] def guessReportLanguage(
       languageOpt: Option[String],
-      report: CoverageReport
+      report: CoverageReport,
+      reportFilePath: String
   ): Either[String, String] = {
     languageOpt match {
       case Some(l) => Right(l)
       case None =>
-        report.fileReports.flatMap(file => Languages.forPath(file.filename)).headOption match {
+        val reportLanguages = getReportLanguages(report).distinct
+        reportLanguages.headOption match {
           case None => Left("Can't guess the report language")
-          case Some(value) => Right(value.name)
+          case Some(language) =>
+            if (reportLanguages.size > 1) {
+              logger.warn(s"""
+                   |Multiple languages detected in $reportFilePath (${reportLanguages.mkString(",")}).
+                   |  This run will report coverage for the $language language.
+                   |  To make sure you are uploading this report for all languages,
+                   |  check https://docs.codacy.com/coverage-reporter/uploading-coverage-in-advanced-scenarios/#multiple-languages
+                   |  for further details.""".stripMargin)
```

Comment:
Suggestion to make the warning message shorter and reinforce that **only** the "elected" language is uploaded:

```suggestion
                   |Multiple languages detected in $reportFilePath (${reportLanguages.mkString(",")}).
                   |  This run will only upload coverage for the $language language.
                   |  To make sure that you upload coverage for all languages in the report,
                   |  see https://docs.codacy.com/coverage-reporter/uploading-coverage-in-advanced-scenarios/#multiple-languages""".stripMargin)
```